common/lib/dependabot/config/file_fetcher.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require "dependabot/file_fetchers/base"
+
+module Dependabot
+  module Config
+    class FileFetcher < Dependabot::FileFetchers::Base
+      CONFIG_FILE_PATHS = %w(.github/dependabot.yml .github/dependabot.yaml).freeze
+
+      def self.required_files_in?(filenames)
+        CONFIG_FILE_PATHS.any? { |file| filenames.include?(file) }
+      end
+
+      def self.required_files_message
+        "Repo must contain either a #{CONFIG_FILE_PATHS.join(' or a ')} file"
+      end
+
+      def config_file
+        @config_file ||= begin
+          cfg = files.first
+          Dependabot::Config::File.parse(cfg.content) if cfg
```

Comment:
Nit: I wonder if it would be clearer to do this parsing step after fetching the file in dry-run? Mostly to line up with existing core concepts.